tests/test_device.py

```diff
@@ -99,20 +116,30 @@ def mock_device(monkeypatch):
         yield Device()
 
 
-
-
 class TestDeviceSupportedLogic:
     """Test the logic associated with the supported operations and observables"""
 
+    # pylint: disable=no-self-use, redefined-outer-name
+
     def test_supports_operation_argument_types(self, mock_device_with_operations):
         """Checks that device.supports_operations returns the correct result
            when passed both string and Operation class arguments"""
 
         assert mock_device_with_operations.supports_operation("PauliX")
         assert mock_device_with_operations.supports_operation(qml.PauliX)
 
-        assert not mock_device_with_operations.supports_operation("PauliY")
-        assert not mock_device_with_operations.supports_operation(qml.PauliY)
+        with pytest.raises(
+                DeviceError,
+                match="Gate {} not supported on device {}".format("S", 'MockDevice'),
+        ):
+            mock_device_with_operations.supports_operation("S")
+
+        with pytest.raises(
+                DeviceError,
+                match="Gate {} not supported on device {}".format(qml.CNOT.__name__,
+                                                                  mock_device_with_operations.short_name),
+        ):
+            mock_device_with_operations.supports_operation(qml.CNOT)
```

Comment:
How come these tests were added? Was this to add coverage?

Reply:
As the functionality of the ``supports_operation`` method was changed, the tests were also updated. Previously, ``supports_operation`` and ``supports_observable`` would return False, if certain conditions were not met, now it raises an error in such cases.

Therefore two of the previous ``assert not`` statements were replaced in each of the tests.